UndertaleModLib/Util/QoiConverter.cs

```diff
@@ -30,19 +30,43 @@ public static class QoiConverter
         /// <param name="s">The stream to create the PNG image from.</param>
         /// <returns>The QOI image as a PNG.</returns>
         /// <exception cref="Exception">If there is an invalid QOIF magic header or there was an error with stride width.</exception>
-        public unsafe static Bitmap GetImageFromStream(Stream s)
+        public static Bitmap GetImageFromStream(Stream s)
         {
-            byte[] header = new byte[12];
-            s.Read(header, 0, 12);
+            Span<byte> header = stackalloc byte[12];
+            s.Read(header);
+            int length = header[8] + (header[9] << 8) + (header[10] << 16) + (header[11] << 24);
+            byte[] bytes = new byte[12 + length];
+            s.Position -= 12;
+            s.Read(bytes, 0, bytes.Length);
+            return GetImageFromSpan(bytes);
+        }
+
+        /// <summary>
+        /// Creates a Bitmap from a <see cref="ReadOnlySpan"/>.
+        /// </summary>
+        /// <param name="bytes">The span to create the PNG image from.</param>
+        /// <returns>The QOI image as a PNG.</returns>
+        /// <exception cref="Exception">If there is an invalid QOIF magic header or there was an error with stride width.</exception>
+        public static Bitmap GetImageFromSpan(ReadOnlySpan<byte> bytes) => GetImageFromSpan(bytes, out _);
+
+        /// <summary>
+        /// Creates a Bitmap from a <see cref="ReadOnlySpan"/>.
+        /// </summary>
+        /// <param name="bytes">The span to create the PNG image from.</param>
+        /// <param name="length">The total amount of data read from the span.</param>
+        /// <returns>The QOI image as a PNG.</returns>
+        /// <exception cref="Exception">If there is an invalid QOIF magic header or there was an error with stride width.</exception>
+        public unsafe static Bitmap GetImageFromSpan(ReadOnlySpan<byte> bytes, out int length)
```

Comment:
Whats the reason for having `length` as out?

Reply:
so that if the passed span is a buffer from a MemoryStream or similar, the current offset could be adjusted by whatever uses the method
i may remove this later in a future PR as it probably won't be needed

Reply:
Is there anything right now that *could* use it?

Reply:
Might as well remove it now then. 🤷